force_nevergrad/mco/ng_mco_communicator.py

```diff
@@ -0,0 +1,90 @@
+#  (C) Copyright 2010-2020 Enthought, Inc., Austin, TX
+#  All rights reserved.
+
+import sys
+import re
+
+from force_bdss.api import (
+    BaseMCOCommunicator,
+    DataValue,
+    FixedMCOParameter,
+    RangedMCOParameter,
+    RangedVectorMCOParameter,
+    ListedMCOParameter,
+    CategoricalMCOParameter
+)
+
+
+class NevergradMCOCommunicator(BaseMCOCommunicator):
+    """ Command-line evaluation.
+
+    Examples
+    --------
+    Evaluate the point (-1.0, 1.0) and write to output.txt,
+
+    echo 1.0,-1.0 | edm run -e force-py36 -- force_bdss
+    --evaluate gaussian.json > output.txt
+
+    Evaluate the point on the first line of input.txt and write to output.txt,
+
+    Windows-native:
+    type input.txt | edm run -e force-py36 -- force_bdss
+    --evaluate gaussian.json > output.txt
+
+    Mac/bash:
+    cat input.txt | edm run -e force-py36 -- force_bdss
+    --evaluate gaussian.json > output.txt
+
+    If you are in the EDM force-py36 environment, then just leave out the
+    'edm run -e force-py36 --' part of the command.
+
+    Notes
+    -----
+    Can only evaluate a single point from stdin. If you want to evaluate
+    multiple points listed in a file, then you must write a script
+    that loops through the file's lines (one line per point) and evaluates
+    each one in turn. cf. evalf.cmd for a windows-native script.
+    """
+
+    def receive_from_mco(self, model):
+        """ Get the parameter values to evaluate from stdin.
+        """
+        # Read in a line of points to evaluate.
+        # Can be tab or comma delimited.
+        line = sys.stdin.readline()
+        data = re.split(r'[,\s]+', line)
+
+        # Get the parameter values based on parameterization and stdin
+        inputs = []
+        for i, param in enumerate(model.parameters):
+
+            # default value of parameter
+            if type(param) == FixedMCOParameter:
+                v = param.value
+            elif type(param) == RangedMCOParameter:
+                v = param.initial_value
+            elif type(param) == RangedVectorMCOParameter:
+                v = param.initial_value
+            elif type(param) == ListedMCOParameter:
+                v = param.levels[0]
+            elif type(param) == CategoricalMCOParameter:
+                v = param.categories[0]
+            else:
+                v = 0.0
+
+            # if the parameter value is specified in the stdin, then take this
+            # and attempt to convert it from string into the appropriate type.
+            try:
+                q = type(v)(data[i])
+                inputs.append(DataValue(value=q, name=param.name))
+            except(ValueError, IndexError):
+                inputs.append(DataValue(value=v, name=param.name))
```

Comment:
Also remember to pass  the `type` attribute here from the parameters into the new `DataValue` instances